sandy/core/endf6.py

```diff
@@ -2475,3 +2475,162 @@ def covariance_info(self, nubar=True, xs=True, mubar=True, chi=True):
+                                            values='VAL')
+
+    def _get_lpc_cov(self):
+        return
+
+    def _get_edistr_cov(self):
+        return
+
+    def get_cov(self, process_mf=[], mf=None, **kwds_njoy):
+        """
+        Get covariance matrices for nubar/xs/energy distribution/Legendre
+        polynomial coefficients from Endf6 or process with NJOY.
+
+        Parameters
+        ----------
+        process_mf : `int` or `list`, optional
+            DESCRIPTION. The default is [].
+        mf : `int` or `list`, optional
+            DESCRIPTION. The default is None.
+        **kwds_njoy : `dict`
+            Extra arguments for proccessing the covariance matrix with NJOY.
+
+        Returns
+        -------
+        cov : `dict` or `sandy.CategoryCov`
+            Dictionary containing the covariance matrix of different mf data.
+            If only one mf is introduced, the method returns the
+            `sandy.CategoryCov` object directly.
+
+        Examples
+        --------
+        Nubar covariance matrix from endf6 file:
+        >>> endf6 = sandy.get_endf6_file('jeff_33','xs', 922380)
+        >>> xs_cov = sandy.XsCov.from_endf6(endf6)
+        >>> nubar_cov = endf6.get_cov(mf=31)
+        >>> assert (nubar_cov.data.values == xs_cov.loc[(9237, 456), (9237, 456)].values).all().all()
+
+        Xs covariance matrix from endf6 file:
+        >>> endf_xs_cov = endf6.get_cov(mf=33)
+        >>> size = len(endf_xs_cov.data)
+        >>> assert (endf_xs_cov.data.values == xs_cov.iloc[:size, :size].values).all().all()
+
+        Both together:
+        >>> endf_xs_cov = endf6.get_cov(mf=[31, 33])
+        >>> size = len(endf_xs_cov[33].data)
+        >>> assert (endf_xs_cov[33].data.values == xs_cov.iloc[:size, :size].values).all().all()
+        """
+        # Select the mf:
+        if mf is not None:
+            listmf_ = [mf] if isinstance(mf, int) else mf
+        else:
+            listmf_ = list(self.to_series().index.get_level_values("MF")
+                               .intersection([31, 33, 34, 35]))
+        list_process = [process_mf] if isinstance(process_mf, int) else process_mf
+        # Dividing the list information:
+        listmf_ = list(pd.Index(listmf_).difference(pd.Index(list_process)))
+        if len(list_process) != 0:
+            kwds_njoy["nubar"] = True if 31 in list_process else False
+            kwds_njoy["xs"] = True if 33 in list_process else False
+            kwds_njoy["chi"] = True if 34 in list_process else False
+            kwds_njoy["mubar"] = True if 35 in list_process else False
+            out = self.get_errorr(**kwds_njoy)
+            cov = out.get_cov(mf=list_process)
+        if len(listmf_) != 0:
+            cov = {} if len(list_process) == 0 else cov
+            if 31 in listmf_:
+                cov[31] = self._get_xs_cov(mf=[31])
+            if 33 in listmf_:
+                cov[33] = self._get_xs_cov(mf=[33])
```

Comment:
By default mf31 and 33 should be taken from errorr

Reply:
At the moment this branch is not able to do that. For that you need the updates of the branch https://github.com/AitorBengoechea/sandy/tree/errorr_get_cov